app/src/molecules/MoleculeAlert.re

```diff
@@ -0,0 +1,105 @@
+open Atom;
+open Particle;
+
+type severity =
+  | Info
+  | Success
+  | Error;
+
+type t = {
+  severity,
+  message: string,
+};
+
+module Styles = {
+  open Css;
+  open DesignSystem;
+
+  let info =
+    style([
+      backgroundColor(Color.gray(~alpha=0.1, ())),
+      borderWidth(px(1)),
+      borderColor(Color.gray()),
+      borderStyle(solid),
+      borderRadius(px(2)),
+      color(Color.gray()),
+      height(px(40)),
+      display(`flex),
+      alignItems(center),
+    ]);
+
+  let success =
+    info
+    << style([
+         backgroundColor(Color.green(~alpha=0.1, ())),
+         borderColor(Color.green()),
+         color(Color.green()),
+       ]);
+
+  let error =
+    info
+    << style([
+         backgroundColor(Color.bordeaux(~alpha=0.1, ())),
+         borderColor(Color.bordeaux()),
+         color(Color.bordeaux()),
+       ]);
+
+  let alert = severity =>
+    switch (severity) {
+    | Info => info
+    | Success => success
+    | Error => error
+    };
+
+  let close =
+    style([
+      cursor(`pointer),
+      height(px(24)),
+      marginLeft(auto),
+      marginRight(px(8)),
+    ]);
+
+  let icon = style([height(px(24)), Css.margin(px(8))]);
+};
+
+module Template = {
+  [@react.component]
+  let make = (~children, ~severity, ~onClick=_ => ()) =>
+    <div className={Styles.alert(severity)}>
+      <div className=Styles.icon>
+        {
```

Comment:
Not specific to this changeset, but one thing about JSX that I find unergonomic is that it invites to freely interleave markup and normal logic. That's why I usually try to avoid more complex inline code and rather assign the result of such switch statements to variables and use them in the markup.

Reply:
I was wondering about the way you decide what to leave in JSX and what not, this clears things up. Will adapt my code.